sirepo/uri_router.py

```diff
@@ -257,6 +257,10 @@ def _init_uris(app, simulation_db):
     assert _default_route, \
         'missing default route'
     _empty_route = _uri_to_route.en
+    i = set(_uri_to_route.keys()).union(sim_types).intersection(
+        simulation_db.SCHEMA_COMMON.rootRedirectUri.keys(),
+    )
+    assert not i, f'duplicate uris={i}'
```

Comment:
I would add one more assertion,

```py
for x in r:
    assert re.search(r'^[a-z]+$'), \
        f'rootRedirectUri={x} must consist of letters only'
```

I think it is good to be restrictive right now, because the purpose is top level marketing links. By restricting, we ensure the namespace is narrow. The act of widening it should be conscious.
 
I would put this in a separate function just because the current function is getting long.

I would

Reply:
https://github.com/radiasoft/sirepo/pull/2786/commits/1912882dc322db1f33962e87148aeddc9c6ef2d7